### DISCUSSION TITLE
Vercel won't install the pg package (solved)

### DISCUSSION THREAD

**Discussion by roangeorge** [vercel/community#854] [Help] [answered]
Hi,

I'm trying to deploy my NextJS app to Vercel, which uses sequelize to interact with my PostgreSQL database and therefore relies on the pg package . My package.json file includes the pg module, however Vercel doesn't seem to install it.

The serverless function guide does not mention anything about why it may not work(https://vercel.com/guides/why-does-my-serverless-function-work-locally-but-not-when-deployed). Any reasons why it's not installing the pg package?

I've attached a screenshot regarding the error.
![image](https://user-images.githubusercontent.com/112395163/190003559-78c509ea-9a6e-41c9-9c85-3f47ed61d9e3.png)

**Comment by amyegan** [comment, marked as answer]
Hi @roandevs! Sorry you're running into this error. Without knowing more about your setup, it's hard to say exactly what's going wrong here. Are you able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) so I can take a closer look?

**Reply by roangeorge** [reply]
Hey there,

Apologies for not including a minimal reproducible example. In the time from posting this to your reply I managed to find a solution to my issue through some older issues opened on Vercel's page (https://github.com/vercel/ncc/issues/345). 

I will include a minimal reproducible example in-case anyone else is stuck with the same kind of error. Originally Vercel was throwing error logs saying that the pg package was not installed and on a local environment this would never occur. My code was originally written as this:

```
const sequelize = require('sequelize');

const database = new sequelize({
    host: process.env.DB_HOST,
    port: 5432,
    username: process.env.DB_USERNAME,
    password: process.env.DB_PASSWORD,
    database: process.env.DB_NAME,
    dialect: "postgres",
    logging: false,
    dialectOptions: {
        timezone: process.env.db_timezone
    },
    define: {
        freezeTableName: true,
        timestamps: false
    }
});
```

However, thanks to that thread it seems a fix for this is specifying the dialectModule as the pg module directly in the Sequelize object parameters as shown:

```
const database = new sequelize({
    host: process.env.DB_HOST,
    port: 5432,
    username: process.env.DB_USERNAME,
    password: process.env.DB_PASSWORD,
    database: process.env.DB_NAME,
    dialect: 'postgres',
    dialectModule: require('pg'),
    logging: false,
    dialectOptions: {
        timezone: process.env.db_timezone
    },
    define: {
        freezeTableName: true,
        timestamps: false
    }
});
```

I tested this and redeployed my project and it's now fixed.

**Reply by amyegan** [reply]
I'm glad you found a solution. Thank you for coming back here to share it!

**Reply by jorgeivaldi** [reply]
dialectModule: require('pg'),

Was the solution for me too!!

Thanks so much!!
